### PR TITLE
Include captured console output string in NodeToolResult

### PR DESCRIPTION
**Pull request by yifan-c**
- If captured console output is null, the test code does not implement the capture function
- If captured console output is empty, there is no output captured.

### CODE REVIEW

File: src/main/java/org/apache/cassandra/distributed/api/NodeToolResult.java [hunk 1]
```diff
@@ -134,6 +144,30 @@ public Asserts notificationContains(ProgressEventType type, String msg)
             return this; // unreachable
         }
 
+        public Asserts capturedConsoleOutContains(String substring)
```
Comment (dcapwell) [inline src/main/java/org/apache/cassandra/distributed/api/NodeToolResult.java:147]: maybe `stdoutContains` and `stderrContains`?  `capturedConsole` I don't feel is needed in the name

File: src/main/java/org/apache/cassandra/distributed/api/NodeToolResult.java [hunk 2]
```diff
@@ -134,6 +144,30 @@ public Asserts notificationContains(ProgressEventType type, String msg)
...
+        public Asserts capturedConsoleErrContains(String substring)
+        {
+            return capturedConsoleOutputContains(substring, false);
+        }
+
+        private Asserts capturedConsoleOutputContains(String substring, boolean isOut)
+        {
+            String name = isOut ? "capturedConsoleOut" : "capturedConsoleErr";
+            String output = isOut ? capturedConsoleOut : capturedConsoleErr;
+            AssertUtils.assertNotNull(name + " not defined", output);
```
Comment (dcapwell) [inline src/main/java/org/apache/cassandra/distributed/api/NodeToolResult.java:161]: can only use these if-and-only-if the C* version supports capturing; fair as we can't throw `org.junit.AssumptionViolatedException` to skip the test

File: src/main/java/org/apache/cassandra/distributed/api/NodeToolResult.java [hunk 3]
```diff
@@ -134,6 +144,30 @@ public Asserts notificationContains(ProgressEventType type, String msg)
...
+            AssertUtils.assertFalse("Found no " + name, output.isEmpty());
```
Comment (dcapwell) [inline src/main/java/org/apache/cassandra/distributed/api/NodeToolResult.java:162]: do we care about that, `""` does not contain `"match"` so the fail should be fine

File: src/main/java/org/apache/cassandra/distributed/api/NodeToolResult.java [hunk 4]
```diff
@@ -37,12 +37,22 @@
     private final List<Notification> notifications;
     private final Throwable error;
 
+    public final String capturedConsoleOut;
```
Comment (dcapwell) [inline src/main/java/org/apache/cassandra/distributed/api/NodeToolResult.java:40]: make these private with a getter to match the rest of the code

File: src/main/java/org/apache/cassandra/distributed/api/NodeToolResult.java [hunk 5]
```diff
@@ -65,6 +74,16 @@ public Throwable getError()
         return error;
     }
 
+    public String getStdOut()
```
Comment (dcapwell) [inline src/main/java/org/apache/cassandra/distributed/api/NodeToolResult.java:77]: for consistency, can we use `getStdout` or update asserts to be camel cased with the O?

File: src/main/java/org/apache/cassandra/distributed/api/NodeToolResult.java [hunk 6]
```diff
@@ -65,6 +74,16 @@ public Throwable getError()
...
+    {
+        return capturedConsoleOut;
+    }
+
+    public String getStdErr()
```
Comment (dcapwell) [inline src/main/java/org/apache/cassandra/distributed/api/NodeToolResult.java:82]: came comment as above, asserts are `stderr` but this is `stdErr`, should pick one and stick with it.

File: src/main/java/org/apache/cassandra/distributed/api/NodeToolResult.java [hunk 7]
```diff
@@ -134,6 +153,29 @@ public Asserts notificationContains(ProgressEventType type, String msg)
...
+        }
+
+        public Asserts stderrContains(String substring)
+        {
+            return capturedConsoleOutputContains(substring, false);
+        }
+
+        private Asserts capturedConsoleOutputContains(String substring, boolean isOut)
+        {
+            String name = isOut ? "capturedConsoleOut" : "capturedConsoleErr";
```
Comment (dcapwell) [inline src/main/java/org/apache/cassandra/distributed/api/NodeToolResult.java:168]: can we use `stdout` and `stderr` here rather than capturedConsole?

**Comment by yifan-c** [comment]
Thanks! Variable names updated.

**Pull request closed without merging**